### DISCUSSION TITLE
Extend showon to Checkboxes form field

### DISCUSSION THREAD

**Discussion by robbiejackson** [joomla/joomla-cms#47140] [2. Cold Features]
This is a feature request. 

In [Enabling showon to work in list options in JForm (resurrected)](https://github.com/joomla/joomla-cms/pull/18998 "Enabling showon in list options") the `showon` capability was extended to options within an HTML select element, so that the options which appear are conditional upon the value set in another field.

This suggestion is to extend this functionality to the [Checkboxes](https://docs.joomla.org/Checkboxes_form_field_type "Checkboxes") form field type, so that the checkboxes field could include 

`<option value="1" showon="targeta:1">Option 1 text</option>`

and then this checkbox would be shown only if the controlling field had that value.

When the user changed the selection in the controlling field, then the set of checkboxes displayed would have to be changed via javascript. 

The justification for this request is that this situation often arises in web forms, and having this feature would obviate the need to perform an ajax request and rebuild the set of checkboxes manually.

**Comment by ghost** [comment]
move to "Discussions".
